app/src/main/java/com/lucascabral/marvelsuperheroes/presenter/di/NetworkModule.kt

```diff
@@ -0,0 +1,63 @@
+package com.lucascabral.marvelsuperheroes.presenter.di
+
+import com.lucascabral.marvelsuperheroes.extension.md5
+import com.lucascabral.marvelsuperheroes.helper.API_KEY
+import com.lucascabral.marvelsuperheroes.helper.PRIVATE_KEY
+import com.lucascabral.marvelsuperheroes.network.MarvelService
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import org.koin.core.qualifier.named
+import org.koin.dsl.module
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.Retrofit
+import java.util.*
+
+val networkModule = module {
+
+    single(named("BASE_URL")) { "http://gateway.marvel.com/v1/public/" }
```

Comment:
Chegou a tentar usar o `BuildConfig` pra salvar essa String de URL? Comentei num outro ponto do código sobre os (Build Config Fields)[https://developer.android.com/studio/build/gradle-tips#share-custom-fields-and-resource-values-with-your-app-code], fica a recomendação de novo de tentar, sempre que possível, manter essas Strings importantes de uso geral ali. Mas dessa forma também ficou bem boa, só uma recomendação mesmo.

https://medium.com/@ramonrabello/best-practices-for-working-with-different-build-types-c5cd08b1adc6

Como comentei no outro ponto do código, não é o nosso caso aqui de ter build types e rotas diferentes, de `debug`, `release`, `homolog`, enfim. Então essas vantagens de aplicar a BASE_URL no BuildConfig acabam não sendo tão grandes pra nós. Mas recomendo mais por ser um conhecimento interessante e uma boa prática em projetos futuros. :wink:

Reply:
Entendido.